### PR TITLE
Pull Request: Battleships Game logic

### PR DESCRIPTION
**Pull request by johnsonsirv**
#### Request to Merge <====
[Battleships](http://en.battleship-game.org/) game of strategy built with modern vanilla Javascript. It is played on ruled grids (paper or board) on which each player's fleet of ships (including battleships) are marked. The locations of the fleets are concealed from the other player. Players alternate turns calling "shots" at the other player's ships, and the objective of the game is to destroy the opposing player's fleet.
#### See [Live Demo](https://battleships-game.netlify.com/)
***Basic Tests***
Tests cover basic features to validate ``Computer AI player``, ``Ship`` and ``Gameboard``.

**ES6 modules:** Used to separate each page and dynamically inject content to the DOM. This is helpful because it allows each part of the app to be maintained separately and independently.

**Webpack**: used to bundle entire app into a single module and perform production-based optimisations

**PS: I am working alone on this project and the Student Success Team is aware.**

***This project was fun to build**

Thanks in advance. Happy Reviewing

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
+import './scss/battleship.scss';
+import UI from './gameui';
+import GameBoard from './gameboard';
+import AI from './ai';
+
+let gameOver = false;
+UI.generateBoard();
+
+// Place ships on waters
```
Comment (bellom) [inline src/index.js]: **Kindly remove commented code, you can always use git history to access previous code right :slightly_smiling_face:**
Comments: It’s better to delete commented code. If you need to review a previous version of your code, use git history ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

File: src/gameui.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+/* eslint-disable no-alert */
```
Comment (bellom) [inline src/gameui.js]: **Can all inline eslint rule be removed, please define eslint rule in the eslint file.**

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @johnsonsirv  :clap: 
#### Awesome job so far!! :+1: 
#### You have done wonderfully well, you passed `almost` all the checklist for this project, all `jest` test passed :+1:, just few minor issue left........... :slightly_smiling_face: 

**Kindly checkout the comments below !!**

**Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....**

Best,

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

File: src/index.js (hunk 1 above)
Comment (johnsonsirv) [reply src/index.js]: Thank you @bellom so far on my code reviews. I appreciate your prompt responses. 
Correct me if I am wrong, but I was thinking that this reference to deleting a comment is for only ``commented code`` probably, previously working/non-working code, and not entirely for minor descriptive inline comments like ``Game Loop``, ``Change Player Turn``, ``Place ships on waters`` etc.

However, I have removed the comments as you suggested.

File: src/gameui.js (hunk 2 above)
Comment (johnsonsirv) [reply src/gameui.js]: Yes. Thanks for pointing out

**Review by bolah2009 — APPROVED** [review]
It's a 👍🏾 from me!

### :heavy_check_mark: Approved!

Cheers
[Bola](https://bolabuari.com/)

**Comment by johnsonsirv** [comment]
Thank you @bolah2009 for the review.

**Pull request merged**